crypto/pkcs11key/key.go

```diff
@@ -198,10 +241,28 @@ func (ps *Key) getPrivateKey(module ctx, session pkcs11.SessionHandle, label str
 	}
 	privateKeyHandle := objs[0]
 
+	attributes, err := module.GetAttributeValue(session, privateKeyHandle, []*pkcs11.Attribute{
+		pkcs11.NewAttribute(pkcs11.CKA_KEY_TYPE, false),
+	})
+	if err != nil {
+		return noHandle, err
+	}
+	for _, attribute := range attributes {
```

Comment:
This version of the loop will now examine each attribute, and error out if any attribute is other than CKK_RSA or CKK_EC. I was thinking more along the lines of: check that `len(attributes) > 0`, then look just at `attributes[0]`. I think in the common case where there's just one attribute matching CKA_KEY_TYPE it's the same, but that makes the code a little clearer.